routes/register.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const datastore = require('../modules/datastore.js');
+const multer  = require('multer');
+
+const USER_KIND = "User";
+
+async function isValid(username) {
```

Comment:
other interesting validity checks:
- empty string
- string with just spaces
etc.

sanitization is another related area that we could look into [[1](https://linguinecode.com/post/validate-sanitize-user-input-javascript)]
e.g. if the user enters something like `foo').delete()` as their username will it be parsed into a string correctly

Reply:
thank you for these suggestions! i created a ticket to add more robust validation and i'll add a todo.